scipy/stats/stats.py

```diff
@@ -2306,6 +2311,186 @@ def zmap(scores, compare, axis=0, ddof=0):
+
+    The concept for this function is borrowed heavily from the `numpy.histogram`
+    helper function `_hist_optim_numbins_estimator`.
+    """
+    assert isinstance(scale, string_types)
+    # private function should not be called otherwise
+
+    # See https://en.wikipedia.org/wiki/Robust_measures_of_scale
+    scale_conversions = {'raw': 1.0,
+                         'normal': special.erfinv(0.5) * 2.0 * math.sqrt(2.0)}
+    try:
+        return scale_conversions[scale.lower()]
+    except KeyError:
+        raise ValueError("{0} not a valid scale for `iqr`".format(scale))
+
+
+def iqr(x, axis=None, range=(25, 75), scale='raw', nan_policy='propagate',
+           interpolation='linear', keepdims=False):
+    """
+    Compute the interquartile range of the data along the specified axis.
+
+    The interquartile range (IQR) is the difference between the 75th and 25th
+    percentile of the data. It is a measure of the dispersion similar to
+    standard deviation or variance, but is much more robust against outliers
+    [3]_.
+
+    The ``range`` parameter allows this function to compute other percentile
+    ranges than the actual IQR. For example, setting ``range=(0, 100)`` is
+    equivalent to ``numpy.ptp``.
+
+    The IQR of an empty array is scalar zero.
+
+    .. versionadded:: 0.18.0
+
+    Parameters
+    ----------
+    x : array_like
+        Input array or object that can be converted to an array.
+    axis : int or sequence of int, optional
+        Axis along which the range is computed. The default is to compute the
+        IQR for the entire array.
+    range : Two-element sequence containing floats in range of [0,100]
+        Percentiles over which to compute the range. Each must be between 0 and
+        100, inclusive. The default is the true IQR: (25, 75). The order of the
+        elements is not important.
+    scale : scalar or str
+        The numerical value of scale will be divided out of the final result.
+        The following string values are recognized:
+
+          'raw' : No scaling, just return the raw IQR.
+          'normal' : Scale by :math:`2 \\sqrt{2} erf^{-1}(\\frac{1}{2})`, which
+              is approximately `1.349`.
+
+        The default is 'raw'. Array-like scale is also allowed, as long as it
+        broadcasts correctly to the output such that ``out / scale`` is a valid
+        operation. The output dimensions depend on the input array, `x`, the
+        `axis` argument, and the `keepdims` flag.
+    nan_policy : {'propagate', 'raise', 'omit'}, optional
+        Defines how to handle when input contains nan. 'propagate' returns nan,
+        'raise' throws an error, 'omit' performs the calculations ignoring nan
+        values. Default is 'propagate'.
+    interpolation : {'linear', 'lower', 'higher', 'midpoint', 'nearest'}
+        This optional parameter specifies the interpolation method to use
+        when the percentile boundaries lie between two data points `i` and `j`:
```

Comment:
didn't build the docs, but I think this needs a blank line for the list to render correctly